pageserver/src/tenant/storage_layer.rs

```diff
@@ -129,17 +193,29 @@ impl ValuesReconstructState {
             keys_with_image_coverage: None,
             layers_visited: 0,
             delta_layers_visited: 0,
+            io_concurrency: {
+                static IO_CONCURRENCY: once_cell::sync::Lazy<String> =
+                    once_cell::sync::Lazy::new(|| {
+                        std::env::var("NEON_PAGESERVER_VALUE_RECONSTRUCT_IO_CONCURRENCY")
+                            .unwrap_or_else(|_| "serial".to_string())
+                    });
```

Comment:
Ref. our discussion last week, it would be nice if we could randomly vary this in tests -- it would give us a ton of test coverage for ~free.

It seems like we don't currently have any test coverage at all for `Parallel`? Let's make sure we have at least some coverage -- either by randomly enabling it, or manually doing so in a few appropriate tests.

Reply:
I've tweaked a unit test in https://github.com/neondatabase/neon/pull/9353/commits/dba696885d151c444e20620db3af3d5b15468c99 to get at least some coverage.